C/services/common/include/python_plugin_handle.h

```diff
@@ -17,19 +17,91 @@
 #include <plugin_handle.h>
 #include <Python.h>
 
+typedef void* (*pluginResolveSymbolFn)(const char *);
+typedef void (*pluginCleanupFn)();
+
 /**
- * The PythonPluginHandle class is used to represent an interface to 
+ * The PythonPluginHandle class is the base class used to represent an interface to
  * a plugin that is available as a python script
  */
 class PythonPluginHandle : public PluginHandle
 {
 	public:
-		PythonPluginHandle(const char *name, const char *path);
-		~PythonPluginHandle();
-		void *GetInfo();
-		void *ResolveSymbol(const char* sym);
+		// Base constructor
+		PythonPluginHandle(const char *name, const char *path) : m_name(name) {};
+		/**
+		 * Base destructor
+		 *    - Call cleanup on python plugin interface
+		 *    - Close python plugin interface library handle
+		 */
+		~PythonPluginHandle()
+		{
+			pluginCleanupFn cleanupFn =
+				(pluginCleanupFn) dlsym(m_hndl, "PluginInterfaceCleanup");
```

Comment:
Use of "m_hndl" seems out of place here. I understand this would be initialized in derived classes' constructor. May be move this cleanup code also there.

Reply:
Cleanup code is done in base class PythonPluginHandle.

Derived classes might add their own task.

This way we don't duplicate the cleanupcode